be/src/exprs/cast_expr.cpp

```diff
@@ -1127,6 +1134,47 @@ class VectorizedCastExpr final : public Expr {
         }
         return result_column;
     };
+
+    bool is_compilable() const override {
+        return !AllowThrowException && IRHelper::support_jit(FromType) && IRHelper::support_jit(ToType);
+    }
+
+    StatusOr<LLVMDatum> generate_ir_impl(ExprContext* context, const llvm::Module& module, llvm::IRBuilder<>& b,
+                                         const std::vector<LLVMDatum>& datums) const override {
+        auto* l = datums[0].value;
+
+        if constexpr (FromType == TYPE_JSON || ToType == TYPE_JSON) {
+            return Status::NotSupported("JIT casting does not support JSON");
```

Comment:
is_complex_type, like json, object, map, array, struct, binary ? pls check all

Reply:
The ﻿is_compilable() function is designed to filter these types. Here, the ﻿if constexpr is utilized for the purpose of compilation.